server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/ExtremeAggrResult.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.iotdb.tsfile.read.common.BatchData;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public class ExtremeAggrResult extends AggregateResult {
+
+  public ExtremeAggrResult(TSDataType dataType) {
+    super(dataType, AggregationType.EXTREME);
+    reset();
+  }
+
+  public Object getAbsValue(Object v) {
+    double doubleValue;
+    float floatValue;
+    int intValue;
+    long longValue;
+
+    switch (resultDataType) {
+      case DOUBLE:
+        doubleValue = (Double) v;
+        return Math.abs(doubleValue);
+      case FLOAT:
+        floatValue = (Float) v;
+        return Math.abs(floatValue);
+      case INT32:
+        intValue = (Integer) v;
+        return Math.abs(intValue);
+      case INT64:
+        longValue = (Long) v;
+        return Math.abs(longValue);
+      default:
+        throw new UnSupportedDataTypeException(String.valueOf(resultDataType));
+    }
+  }
+
+  public Comparable<Object> compareAbsVal(Comparable<Object> minVal, Comparable<Object> maxVal) {
+    Comparable<Object> extVal;
+
+    Comparable<Object> absMaxVal = (Comparable<Object>) getAbsValue(maxVal);
+    Comparable<Object> absMinVal = (Comparable<Object>) getAbsValue(minVal);
+
+    if (absMaxVal.compareTo(absMinVal) >= 0) {
+      extVal = maxVal;
+    } else {
+      extVal = minVal;
+    }
+    return extVal;
+  }
+
+  @Override
+  public Object getResult() {
+    return hasCandidateResult() ? getValue() : null;
+  }
+
+  @Override
+  public void updateResultFromStatistics(Statistics statistics) {
+    Comparable<Object> maxVal = (Comparable<Object>) statistics.getMaxValue();
+    Comparable<Object> minVal = (Comparable<Object>) statistics.getMinValue();
+
+    Comparable<Object> extVal = compareAbsVal(minVal, maxVal);
+    updateResult(extVal);
+  }
+
+  @Override
+  public void updateResultFromPageData(BatchData dataInThisPage) {
+    updateResultFromPageData(dataInThisPage, Long.MIN_VALUE, Long.MAX_VALUE);
+  }
+
+  @Override
+  public void updateResultFromPageData(BatchData dataInThisPage, long minBound, long maxBound) {
+    Comparable<Object> extVal = null;
+
+    while (dataInThisPage.hasCurrent()
+        && dataInThisPage.currentTime() < maxBound
+        && dataInThisPage.currentTime() >= minBound) {
+      Comparable<Object> currentValue = (Comparable<Object>) dataInThisPage.currentValue();
+      Comparable<Object> absCurrentValue = (Comparable<Object>) getAbsValue(currentValue);
+
+      if (extVal == null) {
+        extVal = currentValue;
+      } else {
+        Comparable<Object> absExtVal = (Comparable<Object>) getAbsValue(extVal);
+        if (absExtVal.compareTo(absCurrentValue) < 0) {
+          extVal = currentValue;
+        } else if (absExtVal.compareTo(absCurrentValue) == 0) {
+          if (extVal.compareTo(currentValue) > 0) {
+            extVal = currentValue;
+          }
+        }
+      }
+
+      dataInThisPage.next();
+    }
+    updateResult(extVal);
+  }
```

Comment:
Please see `MaxValueAggrResult.java` to rewrite this method. Your code logic is confusing.